web/profiles/custom/os2loop/modules/os2loop_oembed/src/TwigExtension/TwigExtension.php

```diff
@@ -0,0 +1,254 @@
+
+  /**
+   * The Messenger service.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * The serializer service.
+   *
+   * @var \Symfony\Component\Serializer\SerializerInterface
+   */
+  protected $serializer;
+
+  /**
+   * Drupal\Core\Render\RendererInterface definition.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * The settings.
+   *
+   * @var \Drupal\os2loop_settings\Settings
+   */
+  protected $settings;
+
+  /**
+   * TwigExtension constructor.
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The http client.
+   * @param \Drupal\media\OEmbed\UrlResolverInterface $urlResolver
+   *   The Url resolver service.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger service.
+   * @param \Symfony\Component\Serializer\SerializerInterface $serializer
+   *   The serializer service.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   Drupals renderer service.
+   * @param \Drupal\os2loop_settings\Settings $settings
+   *   Loop settings.
+   */
+  public function __construct(ClientInterface $http_client, UrlResolverInterface $urlResolver, MessengerInterface $messenger, SerializerInterface $serializer, RendererInterface $renderer, Settings $settings) {
+    $this->httpClient = $http_client;
+    $this->urlResolver = $urlResolver;
+    $this->messenger = $messenger;
+    $this->serializer = $serializer;
+    $this->renderer = $renderer;
+    $this->settings = $settings;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFilters(): array {
+    return [
+      new TwigFilter('create_video', [$this, 'createVideo']),
+    ];
+  }
+
+  /**
+   * Render a video from an embed url or iframe.
+   *
+   * @param array $text
+   *   The input text that the filter was applied to.
+   *
+   * @return \Drupal\Component\Render\MarkupInterface
+   *   The rendered html.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   *   An exception if guzzle fails.
+   * @throws \Exception
+   *   The genral exception.
+   */
+  public function createVideo(array $text): MarkupInterface {
+    // Set string depending on field type.
+    $string = isset($text['#plain_text']) ? $text['#plain_text'] : $text['#context']['value'];
+    if (!empty($this->findIframe($string))) {
+      $string = $this->handleIframe($string);
+    }
+    $videoArray = $this->createVideoFromUrl($string);
+    $videoArray = $this->applyCookieConsent($videoArray);
+
+    $render = [
+      '#theme' => 'os2loop_video_iframe',
+      '#video' => $videoArray,
+    ];
+
+    return $this->renderer->render($render);
+  }
+
+  /**
+   * List of allowed providers.
+   *
+   * @var array
+   *   The allowed providers.
+   */
+  private const ALLOWED_PROVIDERS = [
+    'vimeo.com' => [
+      'type' => 'Oembed',
+      'provider' => 'Vimeo',
+      'requiredCookies' => 'cookie_cat_statistic',
```

Comment:
Defining this on the provider is a really great idea! `requiredCookies` should be an array, but that's a minor detail.

Reply:
For now we are only running with cookie information and Vimeo. Having more than one required cookie for cookie information requires some rewriting of the output that is not really needed currently, so we should lie this to rest for now.